Assets/ModScripts/Submodules/UnscrewMaze.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Linq;
+using UnityEngine;
+
+public class UnscrewMaze : Puzzle
+{
+
+    readonly string[] maze = { "1", "13", "23", "12", "13", "23",
+                               "12", "3", "02", "0", "2", "02",
+                               "012", "3", "012", "13", "013", "03",
+                               "01", "123", "013", "13", "23", "2",
+                               "23", "02", "1", "123", "03", "02", 
+                               "01", "03", "1", "013", "13", "03" };
+    readonly int[] positions;
+    int curPos;
+    readonly bool[] bulbsSolved = { false, false };
+
+    public UnscrewMaze(CruelModkitScript Module, int ModuleID, ComponentInfo Info, byte Components) : base(Module, ModuleID, Info, Components)
+    {
+        Debug.LogFormat("[The Cruel Modkit #{0}] Solving Unscrew Maze.", ModuleID);
+        Debug.LogFormat("[The Cruel Modkit #{0}] Morse characters are {1}. ", ModuleID, Info.Morse);
+
+        positions = Base36ToDec(Info.Morse);
+        Debug.LogFormat("[The Cruel Modkit #{0}] The starting position is ({1}, {2}).", ModuleID, Math.Floor(positions[0] / 6f)+1, (positions[0] % 6) + 1);
+        Debug.LogFormat("[The Cruel Modkit #{0}] Bulb 1's coordinate is ({1}, {2}) and Bulb 2's coordinate is ({3}, {4}).", ModuleID, Math.Floor(positions[1] / 6f) + 1, (positions[1] % 6) + 1, Math.Floor(positions[2] / 6f) + 1, (positions[2] % 6) + 1);
+        Debug.LogFormat("[The Cruel Modkit #{0}] The center button is {1}.", ModuleID, Info.Arrows[8] == 9 ? "white. Use the arrow directions to navigate" : "grey. Use the arrow colors to navigate");
+        
+        curPos = positions[0];
+    }
+
+
+    public override void OnArrowPress(int Arrow)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.Audio.PlaySoundAtTransform(Module.ArrowSounds[Arrow].name, Module.transform);
+        Module.Arrows[Arrow].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
```

Comment:
```suggestion
        if (!Module.IsSolving() && !Module.CheckValidComponents())
```

Add IsSolving so that it doesn't check this over and over

Reply:
Do you want this to be present in all button interactions for any submodule? So that it won't check for valid components if the module is already being solved

Reply:
Yes.